src/prefect/_internal/compatibility/deprecated.py

```diff
@@ -272,3 +273,47 @@ def callback(_):
     DEPRECATED_MODULE_ALIASES.append(
         AliasedModuleDefinition(old_name, new_name, callback)
     )
+
+
+class AsyncCompatProxy(wrapt.ObjectProxy):
+    """
+    A proxy object that allows for awaiting a method that is no longer async.
+
+    See https://wrapt.readthedocs.io/en/master/wrappers.html#object-proxy for more
+    """
+
+    def __init__(self, wrapped):
+        super().__init__(wrapped)
+        self._self_already_awaited = False
+
+    def __await__(self):
+        if not self._self_already_awaited:
+            warnings.warn(
+                "This method is no longer async. Awaiting is not necessary - please remove the `await` keyword.",
```

Comment:
Is it possible to use the name of the wrapped function here? It would make the warning easier to grok.

Also, can we add a date in ~6 months that we'll remove this, like our other deprecations?

Reply:
updated in [e112344](https://github.com/PrefectHQ/prefect/pull/14197/commits/e112344a950c00113253b91778939d9165f23978) so that it now looks like
```python
11:51:58.576 | INFO    | Flow run 'brilliant-urchin' - Submitting task get_random_pokemon to thread pool executor...
/Users/nate/github.com/prefecthq/prefect/flows/proofs/backward_compat_submit.py:20: DeprecationWarning: The 'submit' method on 'Task' is no longer async and awaiting it will raise an error after Dec 2024 - please remove the `await` keyword.
  future1 = await get_random_pokemon.submit(None)
```